resources.Groups.department.ReferenceExample.ps1

```diff
@@ -24,102 +131,206 @@ catch {
+                [void]$missingFields.Add($requiredField)
+                if ($debug -eq $true) { Write-Warning "Resource object is missing required field [$requiredField]" }
+            }
+
+            if ([String]::IsNullOrEmpty($resource.$requiredField)) {
+                $incompleteResource = $true
+                [void]$missingFields.Add($requiredField)
+                if ($debug -eq $true) { Write-Warning "Resource object has a null or empty value for required field [$requiredField]" }
             }
+        }
+
+        if (-Not($incompleteResource -eq $True)) {
+            try {
+                # The names of security principal objects can contain all Unicode characters except the special LDAP characters defined in RFC 2253.
+                # This list of special characters includes: a leading space a trailing space and any of the following characters: # , + " \ < > 
+                # A group account cannot consist solely of numbers, periods (.), or spaces. Any leading periods or spaces are cropped.
+                # https://docs.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2003/cc776019(v=ws.10)?redirectedfrom=MSDN
+                # https://www.ietf.org/rfc/rfc2253.txt
 
-            $distinguishedName = "CN=$($ADGroupParams.Name),$($ADGroupParams.Path)"
-            $groupExists = [bool](Get-ADGroup -Filter { DistinguishedName -eq $distinguishedName } -Server $pdc)
+                # Example: Department (department differs from other objects as the property for the name is "DisplayName", not "Name")
+                $ADGroupName = ("$adGroupNamePrefix" + "$($resource.DisplayName)" + "$adGroupNameSuffix")
+                $ADGroupDescription = ("$adGroupDescriptionPrefix" + "$($resource.DisplayName)" + "$adGroupDescriptionSuffix")
 
-            # If resource does not exist
-            if ($groupExists -eq $False) {
-                <# Resource creation preview uses a timeout of 30 seconds
-                while actual run has timeout of 10 minutes #>
-                Write-Information "Creating $($distinguishedName)"
+                # Example: Title 
+                # $ADGroupName = ("$adGroupNamePrefix" + "$($resource.Name)" + "$adGroupNameSuffix")
+                # $ADGroupDescription = ("$adGroupDescriptionPrefix" + "$($resource.Name)" + "$adGroupDescriptionSuffix")
 
-                if (-Not($dryRun -eq $True)) {
-                    $NewADGroup = New-ADGroup @ADGroupParams
+                # Example: Custom object (custom object consists of all custom properties)
+                # $ADGroupName = ("$adGroupNamePrefix" + "$($resource.CustomField1)" + "$($resource.CustomField2)" + "$adGroupNameSuffix")
+                # $ADGroupDescription = ("$adGroupDescriptionPrefix" + "$($resource)" + "$adGroupDescriptionSuffix")
 
+                $ADGroupParams = @{
+                    Name           = $ADGroupName
+                    SamAccountName = $ADGroupName
```

Comment:
Dit gaat verkeerd bij namen langer dan 20 chars. Zorg ervoor dat samaccountname altijd de id is (externalid)

Reply:
De max chars voor een groep is 63 chars.
Daarnaast is externalid inderdaad een goed advies/best practice, maar dit gaat in de meeste gevallen helaas niet op.
Wat we wel kunnen doen is de id als de standaard zetten en dan de varianten die het meest gebruikt worden als examples erbij.